/**
 * Copyright 2018-2021 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

syntax = "proto3";

message Block {
  oneof type {
    Vertical vertical = 1;
    Horizontal horizontal = 2;
    Column column = 3;
    Expandable expandable = 4;
    Card card = 7;
  }
  bool allow_empty = 5;

  // If this is set, then the Block is a "form", and any widgets
  // contained within will defer sending their values to Python until
  // the form's Submit button is pressed.
  string form_id = 6;

  message Vertical {
  }

  message Horizontal {
    repeated double weights = 3;
  }

  message Column {
  }

  message Expandable {
    string label = 1;
    bool expanded = 2;
  }

  message Card {
  }

  // Next ID: 8
}
